### PR TITLE
Extract functions from entity model template

### PR DESCRIPTION
**Pull request by gmarziou**
Refactor Angular entity.model.ts.ejs template so that it gets simpler to maintain and test, and also to share common code with React.

- Please make sure the below checklist is followed for Pull Requests.

- [X] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [X] Tests are added where necessary
- [X] Documentation is added/updated where necessary
- [X] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
@gmarziou : FYI, there are some errors in Travis builds

**Comment by gmarziou** [comment]
Thanks for notice, this is due to import path not handling the new entity root folder.

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/entity.model.ts.ejs [hunk 1]
```diff
@@ -16,96 +16,13 @@
...
-import { IUser } from 'app/core/user/user.model';
-<%_ } _%>
-<%_ typeImports.forEach(typeImport => { _%>
-<%- typeImport %>;
+<%
+const variablesWithTypes = generateEntityClientFields(pkType, fields, relationships, dto);
+const typeImports = generateEntityClientImports(relationships, dto);
+const defaultVariablesValues = generateEntityClientFieldDefaultValues(fields);
+%>
+<%_ typeImports.forEach( (importedPath, importedType) => { _%>
```
Comment (deepu105) [inline generators/entity-client/templates/angular/src/main/webapp/app/entities/entity.model.ts.ejs]: :clap: nice use of map

**Comment by deepu105** [comment]
@gmarziou I merged it but then noticed that the React model file is not updated to use this. Could you plz update the same there as well?

**Comment by gmarziou** [comment]
Yes I will do it, I thought you wanted to change the way imports are generated in React.

**Comment by gmarziou** [comment]
BTW, React app does not compile when setting "`clientRootFolder": "test-root"` on entities.

**Comment by gmarziou** [comment]
OK, I just looked at React code for entities and it's alien code to me 😄 

I thought the entity model was similar to Angular because I saw also an entity.model.ts.ejs file but it's not included in [generated files](https://github.com/jhipster/generator-jhipster/blob/master/generators/entity-client/files.js#L149-L181).
The model seems to be in entity.tsx.ejs but relationships are not typed.

@deepu105 What should I do?

**Comment by deepu105** [comment]
Oh, yes that part is not finished yet. Ok nevermind, i'll do it when I make
cleanups next week. There are many items I have in pipeline to be done there


Thanks & Regards,
Deepu

**Pull request merged**